### PR TITLE
Add the `Limit` parameter to Job Template Launch Post Request

### LINKED ISSUE #45: Add support for passing the `limit` parameter to AWX jobs
When an AWX job is started it is possible to pass a `limit` parameter to limit the hosts where it will run. Currently the auto-heal service doesn't support this, but it should allow to pass it. For example, to pass the value of the `instance` label of the alert:

```yaml
awxJob:
  template: "My template"
  extraVars:
    myvar: myvalue
    yourvar: yourvalue
  limit: {{ $labels.instance }}
```

To be able to use that `limit` parameter the job template has to be configured with the _Prompt on launch_ flag, so we should also check that it is set.

### PR DESCRIPTION
**Pull request by zgalor**
This is required to resolve https://github.com/openshift/autoheal/issues/45

cc @jhernand @moolitayer

### CODE REVIEW

File: awx/job_template_launch_resource.go [hunk 1]
```diff
@@ -94,6 +95,11 @@ func (r *JobTemplateLaunchPostRequest) ExtraVar(name string, value interface{})
 	return r
 }
 
+func (r *JobTemplateLaunchPostRequest) Limit(value string) *JobTemplateLaunchPostRequest {
```
Comment (jhernand) [inline awx/job_template_launch_resource.go]: Maybe some documentation?

**Review by jhernand — COMMENTED** [review]
Looks good, thanks @zgalor. I think we should start to make documentation mandatory for new patches.

**Comment by zgalor** [comment]
@jhernand Done

File: awx/job_template_launch_resource.go [hunk 2]
```diff
@@ -79,13 +79,16 @@ type JobTemplateLaunchPostRequest struct {
 	Request
 
 	extraVars map[string]interface{}
+	limit     string
 }
 
+// ExtraVars set a map or external variables sent to the awx job
```
Comment (jhernand) [inline awx/job_template_launch_resource.go]: ... AWX job.

File: awx/job_template_launch_resource.go [hunk 3]
```diff
@@ -79,13 +79,16 @@ type JobTemplateLaunchPostRequest struct {
...
 func (r *JobTemplateLaunchPostRequest) ExtraVars(value map[string]interface{}) *JobTemplateLaunchPostRequest {
 	r.extraVars = value
 	return r
 }
 
+// ExtraVar adds a single extranal variable to extraVars map
```
Comment (jhernand) [inline awx/job_template_launch_resource.go]: s/extranal/external/

End the sentence with a dot.

File: awx/job_template_launch_resource.go [hunk 4]
```diff
@@ -94,6 +97,12 @@ func (r *JobTemplateLaunchPostRequest) ExtraVar(name string, value interface{})
 	return r
 }
 
+// Limit allows limiting template execution to specific hosts
```
Comment (jhernand) [inline awx/job_template_launch_resource.go]: End the sentence with a dot.

**Review by jhernand — COMMENTED** [review]
Looks good. Just some minor comments about typos and punctuation.

**Comment by zgalor** [comment]
@jhernand Done

**Review by jhernand — APPROVED** [review]
Thanks @zgalor!

**Comment by jhernand** [comment]
@moolitayer can you review/merge, please?

File: awx/job_template_launch_resource.go [hunk 5]
```diff
@@ -79,13 +79,16 @@ type JobTemplateLaunchPostRequest struct {
...
+	limit     string
 }
 
+// ExtraVars set a map or external variables sent to the AWX job.
 func (r *JobTemplateLaunchPostRequest) ExtraVars(value map[string]interface{}) *JobTemplateLaunchPostRequest {
 	r.extraVars = value
 	return r
 }
 
+// ExtraVar adds a single extranal variable to extraVars map.
```
Comment (jhernand) [inline awx/job_template_launch_resource.go]: Just noticed that we still have to s/extranal/external/ here.

Comment (zgalor) [reply awx/job_template_launch_resource.go]: Fixed

**Comment by jhernand** [comment]
By the way @zgalor, would be nice if you can add to the `examples` directory a new example showing how to use the new `Limit` parameter. Doesn't have to be in this pull request.

**Review by moolitayer — APPROVED** [review]
LGTM :+1:

**Comment by moolitayer** [comment]
@zgalor are you planning to address @jhernand's request here or is this good to go?

**Comment by zgalor** [comment]
@moolitayer @jhernand I think I'll leave the examples to a separate PR.
Please merge as is, and add new release tag

**Comment by moolitayer** [comment]
New release in https://github.com/moolitayer/awx-client-go/releases

**Comment by zgalor** [comment]
opened an issue to add example https://github.com/moolitayer/awx-client-go/issues/15

**Pull request merged**